editor/src/components/canvas/canvas-strategies/strategies/reparent-metastrategy.tsx

```diff
@@ -106,6 +123,80 @@ export function getApplicableReparentFactories(
   return factories
 }
 
+function getStartingTargetParentsToFilterOutInner(
+  canvasState: InteractionCanvasState,
+  interactionSession: InteractionSession,
+): ReparentTargetsToFilter | null {
+  if (isInsertionSubjects(canvasState.interactionTarget)) {
+    return null
+  }
+
+  const interactionData = interactionSession.interactionData
+  if (interactionData.type !== 'DRAG') {
+    throw new Error(
+      `getStartingTargetParentsToFilterOut should only be called from a DRAG type interaction, not a ${interactionData.type} type interaction`,
+    )
+  }
+
+  const pointOnCanvas = interactionData.originalDragStart
+  const allowSmallerParent = interactionData.modifiers.cmd
+    ? 'allow-smaller-parent'
+    : 'disallow-smaller-parent'
+
+  const reparentSubjects = reparentSubjectsForInteractionTarget(canvasState.interactionTarget)
+
+  const strictBoundsResult = getReparentTargetUnified(
+    reparentSubjects,
+    pointOnCanvas,
+    interactionData.modifiers.cmd,
+    canvasState,
+    canvasState.startingMetadata,
+    canvasState.startingAllElementProps,
+    'use-strict-bounds',
+    allowSmallerParent,
+  )
+
+  const missingBoundsResult = getReparentTargetUnified(
+    reparentSubjects,
+    pointOnCanvas,
+    interactionData.modifiers.cmd,
+    canvasState,
+    canvasState.startingMetadata,
+    canvasState.startingAllElementProps,
+    'allow-missing-bounds',
+    allowSmallerParent,
+  )
+
+  return reparentTargetsToFilter(strictBoundsResult, missingBoundsResult)
+}
+
+function isCanvasState(
+  value: InteractionCanvasState | InteractionSession,
+): value is InteractionCanvasState {
+  return (value as InteractionCanvasState).startingMetadata != null
+}
+
+const getStartingTargetParentsToFilterOut = memoize(getStartingTargetParentsToFilterOutInner, {
+  maxSize: 5,
```

Comment:
wouldn't we benefit if we increase this number?

Reply:
It has pros and cons. This is enough for someone to drag an element that affects an ancestor 5 levels up. I'm not sure how likely that is in practice tbh. We could certainly increase this to 10 or more.

Reply:
I've increased it to 10